File: 0930.和相同的二元子数组/solution.go
func numSubarraysWithSum(nums []int, goal int) int {
	sum, ans := 0, 0
	mp := make(map[int]int, 3e4)
	mp[0] = 1
	for _, i := range nums {
		sum += i
		if cnt, ok := mp[sum-goal]; ok {
			ans += cnt
		}
		mp[sum]++
	}
	return ans
}

File: 0930.和相同的二元子数组/solution.cpp
class Solution {
public:
    int numSubarraysWithSum(vector<int>& nums, int goal) {
        int sum=0;
        int ans=0;
        unordered_map<int,int> ump;
        ump[0]=1;
        for (int i:nums) {
            sum+=i;
            auto p=ump.find(sum-goal);
            if (p!=ump.end()) {
                ans+=p->second;
            }
            ump[sum]++;
        }
        return ans;
    }
};